lms/djangoapps/shoppingcart/models.py

```diff
@@ -1235,6 +1238,30 @@ class CouponRedemption(models.Model):
     coupon = models.ForeignKey(Coupon, db_index=True)
 
     @classmethod
+    def remove_code_redemption(cls, item, user):
```

Comment:
Can you update this method name to be a bit more explicit. We already have a delete_coupon_redemption() method below so perhaps we should rename this to be remove_code_redemption_from_item()